Initial support for @storybook/server

jonspalmer opened the pull request:
Issue: Added a new "server" framework

## What I did

Added @storybook/server based on @storybook/html.
What's new is that the `configure` method takes a new fetchHtml argument that is responsible for fetching the html from the story.

## How to test

I will followup with a kitchen sink example. 

Usage is something like this:

```javascript
import { configure } from '@storybook/server';
import Stories from '../someStorySource';

const fetchHtml = async function(id, params) {
  const url = new URL('http://localhost:3000/storybook_preview_show');
  url.search = new URLSearchParams({id, ...params} ).toString();

  const response = await fetch(url);
  return  await response.text();
}

configure(fetchHtml, () => Stories, module);
```

## Open Questions/Issues

- My Typescript foo is weak. Could do with some 👀to make sure I got it right
- Tests? There don't seem to be tests for frameworks but maybe I'm missing something.
- Async Story loading? What's the right way to handle this? Probably needs first class support in core.

shilman commented:
Amazing progress @bostanio! I'm guessing this will take some back and forth to get right, so I'll start things off:

1) Each framework has at least one example in `examples/` which provides an end-to-end test. This def needs one!

2) I don't think `configure` should take an extra argument. How about providing a separate function to configure it? (Yes, I get that making it required in configure helps you enforce that it's provided)

```
import { setFetchHtml } from '@storybook/server';
setFetchHtml(someFunction);
```

bostanio commented:
@shilman Working on an example. Exploring the simplest 'app' to use as the backend. I'm guessing its express. 

Can certainly change the `configure` signature. I like forcing it in configure given its required but I can see the value in all frameworks having the same signature.

shilman commented:
I'd also be open to a third `options` object argument to `configure`.

bostanio commented:
> I'd also be open to a third `options` object argument to `configure`.

OK I'll go with that.

bostanio commented:
Added a first pass at the example app. Would love some advice on how we might structure that example and how much of a 'kitchen sink' we want it to be.

tmeasday commented on app/server/src/client/preview/render.ts:
```diff
@@ -0,0 +1,46 @@
+}: RenderMainArgs) {
+  const params = storyFn();
+
+  const element = await fetchStoryHtml(id, params);
```
Should we show a useful error here if the user forgets to set this?

bostanio replied:
Yup that's a great idea. It's entirely unrecoverable.

bostanio replied:
Done

tmeasday commented on app/server/src/server/framework-preset-server.ts:
```diff
@@ -0,0 +1,22 @@
+          test: /\.html$/,
+          use: [
+            {
+              loader: require.resolve('html-loader'),
```
What do we need this for?

bostanio replied:
entirely copy pasta from @storybook/html. I was unclear for the need of them there. We can nuke right?

tmeasday replied:
I think so. This is to allow you to import `.html` files in storybook builds that use this addon. That's not something you would do with `@storybook/server`, right?

tmeasday commented on app/server/src/server/build.ts:
```diff
@@ -0,0 +1,4 @@
+import { buildStatic } from '@storybook/core/server';
+import options from './options';
+
+buildStatic(options);
```
What do these files do? Were they copied over from `@storybook/html`?

bostanio replied:
yup copied. I'm fuzzy on what this stuff is needed for. Should we nuke?

bostanio replied:
Do we need this at all?

shilman replied:
I think we need this:

```
  "bin": {
    "build-storybook": "./bin/build.js",
```

tmeasday reviewed (changes requested):
This is so cool!

bostanio commented on examples/server-kitchen-sink/client/storybook.js:
```diff
@@ -0,0 +1,19 @@
+  url.search = new URLSearchParams(params).toString();
+
+  // eslint-disable-next-line no-undef
+  const response = await fetch(url);
```
What's the right way to fix this without the linting hack?

shilman replied:
https://github.com/eslint/eslint/issues/4015#issuecomment-334929169

bostanio replied:
nice that was an easy fix

shilman commented on app/server/src/client/preview/index.ts:
```diff
@@ -33,10 +32,10 @@ const setRenderFecthAndConfigure: ClientApi['configure'] = (loader, module, opti
-export const forceReRender: ClientApi['forceReRender'] = api.forceReRender;
-export const getStorybook: ClientApi['getStorybook'] = api.clientApi.getStorybook;
-export const raw: ClientApi['raw'] = api.clientApi.raw;
+export const { addDecorator } = api.clientApi;
```
combine into one?

```js
export const { addParameters, clearDecorators, ...etc } = api.clientApi;
```

bostanio replied:
fixed

Commit:
fix: allow tabs to scroll horizontally
fixes #9299

tmeasday commented on app/server/README.md:
```diff
@@ -0,0 +1,25 @@
+# Storybook for Server
```
I guess we might need to expand on this a bit

tmeasday replied:
Will open a separate ticket for this

tmeasday commented on app/server/src/client/preview/index.ts:
```diff
@@ -0,0 +1,43 @@
+  });
+};
+
+const setRenderFecthAndConfigure: ClientApi['configure'] = (loader, module, options) => {
```
```suggestion
const setRenderFetchAndConfigure: ClientApi['configure'] = (loader, module, options) => {
```

tmeasday commented on app/server/src/client/preview/render.ts:
```diff
@@ -0,0 +1,61 @@
+
+const rootElement = document.getElementById('root');
+
+let fetchStoryHtml: FetchStoryHtmlType = async (url, id, params) => {
```
```suggestion
let fetchStoryHtml: FetchStoryHtmlType = async (url, path, params) => {
```

Let's call this path and allow it to be empty (not sure if we should drop the trailing slash in that case?

tmeasday commented on app/server/src/client/preview/render.ts:
```diff
@@ -0,0 +1,61 @@
+    rootElement.innerHTML = element;
+  } else if (element instanceof Node) {
+    // Don't re-mount the element if it didn't change and neither did the story
+    if (rootElement.firstChild === element && forceRender === true) {
```
```suggestion
    if (rootElement.firstChild === element && forceRender !== true) {
```

Is this what you meant?

tmeasday commented on app/server/src/lib/compiler/decorators/knobs.ts:
```diff
@@ -0,0 +1,93 @@
+  const knobParam = param || name; // Todo: can we do away with this?
+  const level = 2;
+  const stringifiedValue = stringifyObject(value, level);
+  // TODO: Add group
```
Do we need to fix this TODO?

tmeasday commented on app/server/src/server/framework-preset-server.ts:
```diff
@@ -0,0 +1,24 @@
+        },
+      ],
+    },
+  };
```
So is the idea that you add the `x.stories.json` files to your storybook directory? What about the idea of fetching the json on startup?

tmeasday replied:
Discussed this and the answer is it is better to let the external process (i.e. Rails) watch the relevant files and regenerate the JSON file, triggering webpack to rebuild when the stories change.

tmeasday reviewed (commented):
Looks great; I have some questions around the loading of story JSON.

shilman committed:
v5.3.5

shilman committed:
Update version.ts to 5.3.5

shilman committed:
5.3.5 latest.json version file

shilman committed:
Core: Fix default sorting of docs-only stories

shilman committed:
Core: Update tests for docs-only stories

shilman committed:
Official-storybook: Don't load intro.stories.mdx

atanasster committed:
bypass source-loader if no exports

shilman committed:
Core: Fix default sorting of docs-only stories (#9504)

Core: Fix default sorting of docs-only stories

shilman committed:
Source-loader: Bypass if file has no exports (#9505)

Source-loader: Bypass if file has no exports

shilman committed:
5.3.6 changelog

shilman committed:
Update peer dependencies to 5.3.6

shilman committed:
v5.3.6

shilman committed:
Update version.ts to 5.3.6

shilman committed:
5.3.6 latest.json version file

jamesgeorge007 committed:
refactor: use startsWith

jamesgeorge007 committed:
deps: update husky to v4

HarveyPeachey committed:
Add BBC to list of examples

shilman committed:
Vue-kitchen-sink add component docgen description

atanasster committed:
remove dedent default and white space wrapping

ndelangen committed:
Add BBC to list of examples (#9510)

Add BBC to list of examples

ndelangen committed:
chore(deps): update husky to v4 (#9509)

chore(deps): update husky to v4

ndelangen committed:
Refactor: use startsWith (#9508)

Refactor: use startsWith

ndelangen committed:
Revert "Add BBC to list of examples"

alexzherdev committed:
Fix typos in "Writing presets"

shilman committed:
Fix typos in "Writing presets" (#9524)

Fix typos in "Writing presets"

atanasster committed:
proptable: convert default prop to string

atanasster committed:
added check to make sure summary is defined

atanasster committed:
revert syntaxhighlighter, added Source.defaultProps.format = false

atanasster committed:
remove white space styling

gaetanmaisse committed:
docs: add a section about external puppeteer in MIGRATION.md

gaetanmaisse committed:
docs(storyshot-puppeteer): add a warning to emphasize the fact that as of SB 5.3 puppeteer must be added to the project dep directly

shilman committed:
Addon-docs: Preserve Source indentation by default (#9513)

Addon-docs: Preserve Source indentation by default

shilman committed:
Addon-docs Props: Convert default prop value to string (#9525)

Addon-docs Props: Convert default prop value to string

shilman committed:
Update addon storyshot puppeteer doc (#9526)

Update addon storyshot puppeteer doc

fireseasonnow committed:
Fix broken link to repo in empty knobs panel

The link to repo in empty knobs panel couldn't be triggered. The problem was with missing `cancel={false}` prop, because `Link` component has a `cancel={true}` by default.

dnielbowen committed:
Enabling HMR logging in browser console

gaetanmaisse committed:
fix: clean usage of `@types/webpack-env` dep in all packages

 - `@types/webpack-env` must be defined as `dependencies` in packages that export a type/function using typings from `@types/webpack-env` (it is often `NodeModule`). In that case `/// <reference types="webpack-env" />` is present in some files of the `dist` folder of the lib. Example: `lib/client-api`.
 - `@types/webpack-env` must be defined as `devDependencies` in packages that use its typings internally but does not expose anything related to it. Many addons need the type because they use `module.hot` function. Example: `addons/a11y`
 - `webpack-env` should be removed from "compilerOptions.types" key of `tsconfig.json` of packages that don't need it.

 I also bump `@types/webpack-env` to ^1.15.0 to have the same version all across the monorepo

gaetanmaisse committed:
fix: move `@types/npmlog` to dependencies as `node-logger` is exporting `npmlog`

As `npmlog` is exported (`export { npmLog as instance };`), TS users of this lib need to have `npmlog` types too, so `@types/npmlog` have to be in `dependencies` and not `devDependencies`

devtronic committed:
Update 1-Button.stories.ts

The property in the Angular component is lowercase. Without this change, the button example story wont work properly.
(The button text isn't rendered)

shilman committed:
Core: Fix legacy story URLs

atanasster committed:
initial check0in

shilman committed:
Core: Fix legacy story URLs (#9545)

Core: Fix legacy story URLs

atanasster committed:
clean up further

atanasster committed:
re-generate test snapshots

atanasster committed:
removed unused code

shilman committed:
Node-logger: Move `@types/npmlog` to dependencies (#9538)

Node-logger: Move `@types/npmlog` to dependencies

shilman committed:
Revert "Add BBC to list of examples" (#9522)

Revert "Add BBC to list of examples"

shilman committed:
Merge branch 'next'

shilman committed:
5.3.7 changelog

shilman committed:
Update peer dependencies to 5.3.7

shilman committed:
v5.3.7

shilman committed:
Update version.ts to 5.3.7

shilman committed:
5.3.7 latest.json version file

ndelangen committed:
Merge branch 'next-6.0.0' into ts-migration/polymer

ndelangen committed:
Merge branch 'next' into next-6.0.0

# Conflicts:
#	addons/cssresources/package.json
#	examples/dev-kits/package.json
#	examples/rax-kitchen-sink/package.json

ndelangen committed:
ADD page about tri-config intro

ndelangen committed:
FIX missing dep

ndelangen committed:
ADD api

ndelangen committed:
ADD migration guide ref to docs-pages that are not a starter-guide

shilman committed:
Clean usage of `@types/webpack-env` dep in all packages (#9536)

Clean usage of `@types/webpack-env` dep in all packages

ndelangen committed:
Next 6.0.0 (#9212)

Next 6.0.0

dependabot-preview[bot] committed:
Bump terser-webpack-plugin from 2.2.1 to 2.3.2

Bumps [terser-webpack-plugin](https://github.com/webpack-contrib/terser-webpack-plugin) from 2.2.1 to 2.3.2.
- [Release notes](https://github.com/webpack-contrib/terser-webpack-plugin/releases)
- [Changelog](https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/terser-webpack-plugin/compare/v2.2.1...v2.3.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

urish committed:
fix(webcomponents): make types play nicely with lit-element

Before this change, returning html`<sometag>...</sometag>` from `storiesOf(...).add` would result in the following error: Type 'TemplateResult' is not assignable to type 'StoryFnHtmlReturnType'

ndelangen committed:
Merge branch 'next' into pr/avasuro/7981

# Conflicts:
#	lib/cli/package.json

ndelangen committed:
Fixed Angular button example story (#9540)

Fixed Angular button example story

atanasster committed:
bolean props default false

shilman committed:
6.0.0-alpha.0 changelog

ndelangen committed:
Having this causes our bootstrap procedure to fail sporadically locally, failing tests, and publish attempts

atanasster committed:
withStorySource deprecation

ndelangen committed:
Merge branch 'next' into core/add-docs-config-overview

ndelangen committed:
FIX lockfile

atanasster committed:
fix parameters, when locationsMap is null

ndelangen committed:
Merge branch 'next' into pr/kroeder/8102

# Conflicts:
#	app/polymer/package.json

ndelangen committed:
REMOVE yarn/releases (#9563)

REMOVE yarn/releases

ndelangen committed:
Add/app marionette js (#7981)

Add/app marionette js

ndelangen committed:
Typescript: Migrate `@storybook/polymer` (#8102)

Typescript: Migrate `@storybook/polymer`

ndelangen committed:
Merge pull request #9229 from storybookjs/core/preview-layouts

Core/preview layouts

ndelangen committed:
Typescript: Migrate `@storybook/polymer (#9565)

Typescript: Migrate `@storybook/polymer

atanasster committed:
api: adding useSharedState, useStoryState

philippe-lavoie committed:
Removing lineHeight hard code.

With lineHeight set to set 1, the bottom of the letters like 'p' gets truncated for most fonts.

atanasster committed:
useStorybookState instead of useContext

ndelangen committed:
Bump terser-webpack-plugin from 2.2.1 to 2.3.2 (#9386)

Bump terser-webpack-plugin from 2.2.1 to 2.3.2

ndelangen committed:
feat(addon-docs): named colors with ColorPalette (#9453)

feat(addon-docs): named colors with ColorPalette

dependabot-preview[bot] committed:
Bump corejs-upgrade-webpack-plugin from 2.2.0 to 3.0.1

Bumps [corejs-upgrade-webpack-plugin](https://github.com/ndelangen/corejs-upgrade-webpack-plugin) from 2.2.0 to 3.0.1.
- [Release notes](https://github.com/ndelangen/corejs-upgrade-webpack-plugin/releases)
- [Commits](https://github.com/ndelangen/corejs-upgrade-webpack-plugin/compare/v2.2.0...v3.0.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

ndelangen committed:
FIX cra-ts-kitchen-sink/src/stories/1-Button.stories.tsx

ndelangen committed:
FIX manager using ESM in dev-kits example

ndelangen committed:
IMPROVE the build-storybooks script

shilman committed:
Merge branch 'next' into prop-table-default-false

shilman committed:
Removing lineHeight hard code. (#9567)

Addon-docs: Remove hard-coded lineHeight in Typeset block

shilman committed:
Revert "Revert "Add BBC to list of examples""

shilman committed:
Addon-docs:  Fix TS false default value in prop table (#9560)

Addon-docs:  Fix TS false default value in prop table

SoloJiang committed:
fix: readme

SoloJiang committed:
chore: remove rax-button

SoloJiang committed:
fix: readme

shilman committed:
CLI: Improve Rax template (#9574)

CLI: Improve Rax template

robcaldecottvelo committed:
Bumped react-dev-utils dependency to v10.

ndelangen committed:
Make build-ScriptDetection sync

ndelangen committed:
Revert "Revert "Add BBC to list of examples"" (#9571)

Revert "Revert "Add BBC to list of examples""

ndelangen committed:
IMPROVE the build-storybooks script (#9569)

IMPROVE the build-storybooks script

ndelangen committed:
Bump corejs-upgrade-webpack-plugin from 2.2.0 to 3.0.1 (#9427)

Bump corejs-upgrade-webpack-plugin from 2.2.0 to 3.0.1

shilman committed:
MDX support is official

ndelangen committed:
Merge branch 'next' into pr/urish/9557

shilman committed:
update mdx syntax placeholder

ndelangen committed:
FIX linting

ndelangen committed:
Merge branch 'next' into patch-webcomponents-types

ndelangen committed:
CLEANUP linting issues

ndelangen committed:
api: adding useSharedState, useStoryState (#9566)

api: adding useSharedState, useStoryState

ndelangen committed:
Bump @babel/runtime from 7.7.4 to 7.7.7 (#9277)

Bump @babel/runtime from 7.7.4 to 7.7.7

ndelangen committed:
Bump babel-plugin-emotion from 10.0.23 to 10.0.27 (#9239)

Bump babel-plugin-emotion from 10.0.23 to 10.0.27

ndelangen committed:
Bump babel-plugin-macros from 2.7.1 to 2.8.0 (#9236)

Bump babel-plugin-macros from 2.7.1 to 2.8.0

shilman committed:
Bumped react-dev-utils dependency to v10. (#9579)

Bumped react-dev-utils dependency to v10.

ndelangen committed:
CHANGE netlify to build examples

ndelangen committed:
REMOVE now

shilman committed:
Fix CSF in MDX recipe

shilman committed:
fix typo

shilman committed:
Fix CSF in MDX recipe (#9587)

Fix CSF in MDX recipe

shilman committed:
6.0.0-alpha.0 changelog updates

shilman committed:
Update peer dependencies to 6.0.0-alpha.0

shilman committed:
v6.0.0-alpha.0

shilman committed:
Update version.ts to 6.0.0-alpha.0

shilman committed:
6.0.0-alpha.0 next.json version file

ndelangen committed:
FIX lockfile & FIX linting

ndelangen committed:
Merge branch 'next' of github.com:storybooks/storybook into next

ndelangen committed:
FIX lockfile & version.ts

ndelangen committed:
Merge branch 'next' of github.com:storybooks/storybook into next

ndelangen committed:
Merge branch 'next' into core/netlify-again

ndelangen committed:
Merge branch 'next' into core/netlify-again

ndelangen committed:
Merge branch 'next' into pr/urish/9557

ndelangen committed:
Merge branch 'next' into patch-webcomponents-types

shilman committed:
Typescript: Export IStory in `@storybook/angular` (#9097)

Typescript: Export IStory in `@storybook/angular`

atanasster committed:
code changes according to shilman comments

ndelangen committed:
Fix broken link to repo in empty knobs panel (#9530)

Fix broken link to repo in empty knobs panel

shilman committed:
Merge branch 'next' into pr/9547

shilman committed:
5.3.8 changelog

ndelangen committed:
REMOVE polymer

It's moved into it's own repo here:
https://github.com/storybookjs/storybook

ndelangen committed:
DELETE references to polymer

jpylisela committed:
Remove duplicate text

Minor update - Remove duplicate text regarding Create React App.

ndelangen committed:
FIX lockfile

ndelangen committed:
DETACH react-native

ndelangen committed:
CLEANUP

ndelangen committed:
FIX lockfile

ndelangen committed:
FIX lockfile

ndelangen committed:
FIX problem with typings after regen lockfile

ndelangen committed:
WIP

ndelangen committed:
FIX

ndelangen committed:
FIX

ndelangen committed:
FIX lockfile

ndelangen committed:
FIX some dependencies & ts problems on next (#9603)

FIX some dependencies & ts problems on next

ndelangen committed:
Merge branch 'next' into remove/polymer

ndelangen committed:
FIX lockfile

ndelangen committed:
Merge branch 'next' into detach/react-native

# Conflicts:
#	yarn.lock

ndelangen committed:
no message

ndelangen committed:
FIX now + netlify

shilman committed:
Remove duplicate text (#9597)

Remove duplicate text

ndelangen committed:
FIX lockfile

shilman committed:
REMOVE polymer (#9596)

REMOVE polymer

shilman committed:
Merge branch 'next' into detach/react-native

shilman committed:
DETACH react-native (#9599)

DETACH react-native

ndelangen committed:
Enabling HMR logging in browser console (#9535)

Enabling HMR logging in browser console

shilman committed:
6.0.0-alpha.1 changelog

shilman committed:
Update peer dependencies to 6.0.0-alpha.1

shilman committed:
v6.0.0-alpha.1

shilman committed:
Update version.ts to 6.0.0-alpha.1

shilman committed:
6.0.0-alpha.1 next.json version file

shilman committed:
Publish: Remove docs to reduce package size

ndelangen committed:
FIX lockfile

shilman committed:
Merge branch 'next' into shilman/remove-docs-from-publish

ndelangen committed:
FIX lockfile

callumacrae committed:
fix(addon-storyshots): Remove excess slashes from jest transform warning

It's printing all of them, so I copied all of them, and that didn't work!

shilman committed:
fix(addon-storyshots): Remove excess slashes from jest transfor… (#9616)

fix(addon-storyshots): Remove excess slashes from jest transform warning

ndelangen committed:
Merge branch 'next' into core/add-docs-config-overview

ndelangen committed:
FIX spelling mistakes

South-Paw committed:
typescript config documentation

South-Paw committed:
indent options to make them stand out better

South-Paw committed:
fix typo

ndelangen committed:
Improve Typescript Configuration documentation page (#9617)

Improve Typescript Configuration documentation page

richardhuf84 committed:
withKnobs const set twice

ndelangen committed:
Merge pull request #9612 from storybookjs/shilman/remove-docs-from-publish

Publish: Remove docs to reduce package size

ndelangen committed:
Merge pull request #9389 from storybookjs/imgbot

[ImgBot] Optimize images

shilman committed:
Merge pull request #9620 from richardhuf84/next

withKnobs const set twice

shilman committed:
5.3.9 changelog

Commit:
Enables the consumer to hide the code

Commit:
Adds minimal documentation to the README

Commit:
Adds basic test

ndelangen committed:
IMPROVE the section about presets-api addons

ndelangen committed:
Merge branch 'next' into core/add-docs-config-overview

ndelangen committed:
MIGRATE ember to new "import { hbs } from 'ember-cli-htmlbars'"

ndelangen committed:
MOVE ember dist to "ember-output"

ndelangen committed:
FIX pebkac problem

ndelangen committed:
MIGRATE ember to new "import { hbs } from 'ember-cli-htmlbars'" (#9633)

MIGRATE ember to new "import { hbs } from 'ember-cli-htmlbars'"

shilman committed:
Clarify and expand on addon-docs manual config

shilman committed:
Clarify and expand on addon-docs manual config (#9638)

Clarify and expand on addon-docs manual config

ndelangen committed:
Update docs/src/pages/presets/writing-presets/index.md

Co-Authored-By: Tom Coleman <tom@thesnail.org>

ndelangen committed:
Core: Document main.js config (#9552)

Core: Document main.js config

Commit:
feat(addon-docs): ColorPalette style

Commit:
feat(addon-docs): ColorPalette style

Commit:
feat(addon-docs): ColorPalette style

ndelangen committed:
fix(addon-docs): ColorPalette style (#9643)

fix(addon-docs): ColorPalette style

rbardini committed:
Add Paddings addon to addon gallery

ndelangen committed:
UPGRADE the mini-css-extract-plugin

ndelangen committed:
Merge branch 'next' into core/netlify-again

ndelangen committed:
Netlify for examples again (#9585)

Netlify for examples again

ndelangen committed:
Add Paddings addon to addon gallery (#9647)

Add Paddings addon to addon gallery

ndelangen committed:
Update bar.tsx

ndelangen committed:
Merge branch 'next' into knob-groups-overflow-not-scrolling

ndelangen committed:
Merge branch 'next' into next

ndelangen committed:
Merge branch 'next' into patch-webcomponents-types

shilman committed:
Merge pull request #9652 from storybookjs/fix/9462-sass-loading

Core: Upgrade `min-css-extract-plugin` to fix SASS loading

shilman committed:
Merge branch 'next' into pr/9090

shilman committed:
Sidebar: Clean up viewMode handling and add tests

shilman committed:
Merge branch 'next' into pr/9090

shilman committed:
Document viewMode story parameter

shilman committed:
UI: Add `viewMode` parameter to control story nav UI (#9090)

UI: Add `viewMode` parameter to control story nav UI

shilman committed:
Merge branch 'next' into pr/9547

shilman committed:
Create source-loader.stories.storyshot

AlbertoLunaPe committed:
Fix default value for web components prop table docs

As `custom-elements.json` format changed we have to
support `default` to get the default value.
This change still keeps compatibility with `defaultValue`.

shilman committed:
Source-loader: Overhaul to remove decorators, support user-conf… (#9547)

Source-loader: Overhaul to remove decorators, support user-configurable source

shilman committed:
Merge branch 'next' into pr/9095

ndelangen committed:
ADD notices describing the incompatibilities with manager.js theming & addon-docs && make the addon-docs README the sot

ndelangen committed:
documentation fix #9498 - theming + addon-docs incompatible (#9661)

documentation fix #9498 - theming + addon-docs incompatible

ndelangen committed:
ADD support for className prop on Form.Field

shilman committed:
Merge pull request #9665 from storybookjs/fix/styled-form-fields

UI: Add support for className prop on Form.Field

shilman committed:
Merge pull request #9383 from baker-travis/knob-groups-overflow-not-scrolling

UI: Fix tabs to scroll horizontally

ndelangen committed:
fix(webcomponents): make types play nicely with lit-element (#9557)

fix(webcomponents): make types play nicely with lit-element

ndelangen committed:
@storybook/addon-cssresources - Hide Code (#9627)

@storybook/addon-cssresources - Hide Code

ndelangen committed:
REMOVE addon-info & addon-notes

They have been MOVED to https://github.com/storybookjs/deprecated-addons

ndelangen committed:
CLEANUP lockfile

ndelangen committed:
REMOVE unneeded snapshots

ndelangen committed:
Merge branch 'next' into remove/addon-notes-info

ndelangen committed:
CLEANUP

ndelangen committed:
Merge branch 'remove/addon-notes-info' of github.com:storybooks/storybook into remove/addon-notes-info

ndelangen committed:
FIX linting

ndelangen committed:
Fix default value for web components prop table docs (#9655)

Fix default value for web components prop table docs

ndelangen committed:
configure tabs: title, visibility, order and disable (#9095)

configure tabs: title, visibility, order and disable

shilman committed:
6.0.0-alpha.2 changelog

shilman committed:
Update peer dependencies to 6.0.0-alpha.2

shilman committed:
v6.0.0-alpha.2

shilman committed:
Update version.ts to 6.0.0-alpha.2

shilman committed:
6.0.0-alpha.2 next.json version file

ndelangen committed:
CHANGE devtools example for previewWrapper

ndelangen committed:
Merge branch 'next' into remove/addon-notes-info

# Conflicts:
#	addons/info/package.json
#	addons/notes/package.json
#	examples/angular-cli/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/cra-ts-kitchen-sink/package.json
#	examples/ember-cli/package.json
#	examples/html-kitchen-sink/package.json
#	examples/mithril-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/official-storybook/stories/addon-notes.stories.js
#	examples/preact-kitchen-sink/package.json
#	examples/rax-kitchen-sink/package.json
#	examples/riot-kitchen-sink/package.json
#	examples/svelte-kitchen-sink/package.json
#	examples/vue-kitchen-sink/package.json
#	examples/web-components-kitchen-sink/package.json

ndelangen committed:
FIX lockfile

atanasster committed:
check if docsOnly is set to hide the addon panels

shilman committed:
MOVE addon-notes & addon-info to own repo (#9673)

MOVE addon-notes & addon-info to own repo

ndelangen committed:
check if docsOnly is set to hide the addon panels (#9687)

check if docsOnly is set to hide the addon panels

shilman committed:
Fix jsconfig support with CRA (#9324)

Fix jsconfig support with CRA

tmeasday committed:
Merge branch 'storybook_server' of https://github.com/jonspalmer/storybook into jonspalmer-storybook_server

tmeasday committed:
Fix typo

tmeasday committed:
Rename `id` to `path` to be a bit more general

tmeasday commented:
@jonspalmer there are still two unresolved comments on that PR that we need to sort out:

1. Is this a mistake or did I misunderstand: https://github.com/storybookjs/storybook/pull/9250#discussion_r366776949

2. Do we need to deal with this TODO: https://github.com/storybookjs/storybook/pull/9250/files/be4d06c0cd69b2ed4a79ad390852770bfd6ccfeb#diff-83aae226cfb73b52d12dfd37ae15a411?

Also a change to `mdx.tsx` got dropped because `generateHrefWithHash` doesn't exist any more. I'm not sure if that's going to be a problem but I'll have a proper play with this once the above two questions are resolved.

ndelangen commented:
@shilman I assume you got a ok from @tmeasday to merge? Looks like he still had a few concerns?

shilman commented:
@ndelangen yeah he filed some follow-up issues

tmeasday commented:
@ndelangen this was a bit confusing but @jonspalmer answered my questions on the other PR
